mlrun/api/api/endpoints/projects.py

```diff
@@ -1,53 +1,45 @@
-from operator import attrgetter
 from http import HTTPStatus
 
 from fastapi import APIRouter, Depends, Response
 from sqlalchemy.orm import Session
 
 from mlrun.api import schemas
 from mlrun.api.api import deps
-from mlrun.api.api.utils import log_and_raise
-from mlrun.api.db.sqldb.helpers import to_dict as db2dict
 from mlrun.api.utils.singletons.db import get_db
 
 router = APIRouter()
 
 
 # curl -d '{"name": "p1", "description": "desc", "users": ["u1", "u2"]}' http://localhost:8080/project
+# TODO: remove the wrong route (/project) before 0.6.0 is out after UI change to new route
 @router.post("/project")
-def add_project(
+@router.post("/projects", response_model=schemas.ProjectOutput)
+def create_project(
     project: schemas.ProjectCreate, db_session: Session = Depends(deps.get_db_session)
 ):
-    project_id = get_db().add_project(db_session, project.dict())
-    return {
-        "id": project_id,
-        "name": project.name,
-    }
+    get_db().create_project(db_session, project)
+    return get_db().get_project(db_session, project.name)
 
 
 # curl -d '{"name": "p1", "description": "desc", "users": ["u1", "u2"]}' -X UPDATE http://localhost:8080/project
+# TODO: remove the wrong route (/project/{name}) before 0.6.0 is out after UI change to new route
 @router.post("/project/{name}")
+@router.put("/projects/{name}", response_model=schemas.ProjectOutput)
 def update_project(
     project: schemas.ProjectUpdate,
     name: str,
     db_session: Session = Depends(deps.get_db_session),
 ):
-    get_db().update_project(db_session, name, project.dict(exclude_unset=True))
-    return {}
+    get_db().update_project(db_session, name, project)
+    return get_db().get_project(db_session, name)
 
 
 # curl http://localhost:8080/project/<name>
-@router.get("/project/{name}", response_model=schemas.ProjectOut)
+# TODO: remove the wrong route (/project/{name}) before 0.6.0 is out after UI change to new route
+@router.get("/project/{name}", response_model=schemas.ProjectOutput)
+@router.get("/projects/{name}", response_model=schemas.ProjectOutput)
 def get_project(name: str, db_session: Session = Depends(deps.get_db_session)):
-    project = get_db().get_project(db_session, name)
-    if not project:
-        log_and_raise(error=f"project {name!r} not found")
-
-    project.users = [u.name for u in project.users]
-
-    return {
-        "project": project,
-    }
+    return get_db().get_project(db_session, name)
 
 
 @router.delete("/projects/{name}", status_code=HTTPStatus.NO_CONTENT.value)
```

Comment:
Delete must go to the `ProjectsManager`, it cannot go just to the MLRun DB. Reason is that if you just delete a project in the DB, `ProjectsManager` will recreate it the next time it runs the periodic syncing, since it still exists on the rest of the consumers.
Therefore, while we can avoid sending all the calls to the `ProjectsManager`, delete must go there.